src/WixToolset.Core.Burn/Bind/BindBundleCommand.cs

```
@@ -243,6 +245,8 @@ public void Execute()
                 }
             }
 
+            duplicateCacheIdDetector = null;
```

This collision detection needs to happen after resolving delayed variables. For example, the `CacheId` can look like this: `CacheId="!(bind.packageDescription.PackageId)"`. https://github.com/wixtoolset/Core/blob/3f4bd928ab4c048792bf4c5c10004a1f22e8aa19/src/WixToolset.Core.Burn/Bind/BindBundleCommand.cs#L367-L371

Assigning to `null` is unnecessary in C#.

---

Putting it right after resolving the delayed fields would make sure that all variables are resolved.

If you want to perform the collision detection at the latest possible moment, that would be when the CacheId is actually written to the Burn manifest: https://github.com/wixtoolset/Core/blob/930b0ca136824d7f4917482fe7a7a577d28f6903/src/WixToolset.Core.Burn/Bundles/CreateBurnManifestCommand.cs#L364

The only real benefit for delaying it like that is to guard against extensions changing the CacheId in the symbol. Most of Core doesn't worry about extensions modifying the symbols like that. The downside to delaying it is that a lot of work is done between those two places and it would be nicer to the user to fail faster.